### DISCUSSION TITLE
Turn off reactions for a specific region

### DISCUSSION THREAD

**Discussion by RSuryaNarayan** [AMReX-Combustion/PeleC#700] [Q&A] [answered]
Hello, 

Is there a way to turn off reactions for a specifc region of interest?

I tried modifying ```advance``` in ```Advance.cpp``` to parse in a ```const amrex::Geometry& geom``` object but I can't seem to get the local ```(i,j,k)``` like in ```pc_init_data```. Wondering if there's a work around for this...

**Comment by marchdf** [comment]
If you aren't scared of a total hack (sounds like you are not), I would put something around here: https://github.com/AMReX-Combustion/PeleC/blob/development/Source/React.cpp#L202 and just test i/j/k for the region where you don't want to call `reactor->react`.

**Reply by RSuryaNarayan** [reply]
Indeed that was exactly what I was thinking about a few minutes ago. @marchdf would playing around with the ```I_R(i,j,k)``` term (i.e. set it zero whereever I don't want reactions) help?

**Comment by hsitaram** [comment]
I am wondering if you can hack the mask variable passed to react call.

**Comment by hsitaram** [comment]
Unfortunately, I see the mask is used only in cvode right now. we may need to update the other react solvers for completeness

**Comment by RSuryaNarayan** [comment]
@hsitaram I am looking for something similar to what you did for the RDE case here: https://www.nrel.gov/docs/fy22osti/82891.pdf
<img width="780" alt="image" src="https://github.com/AMReX-Combustion/PeleC/assets/52670594/5716e2ed-f7d0-4bda-b71d-3a50221c6329">

**Reply by SreejithNREL** [reply, marked as answer]
Hello @RSuryaNarayan, Can you please try this? (https://github.com/SreejithNREL/PeleC/tree/Reaction_Mask_Implementation)
You can specify the box in which chemical mask is activated using the inputs
pelec.chem_mask=1
pelec.lo_chemmask= lo_x lo_y lo_z
pelec.hi_chemmask= hi_x hi_y hi_z.

Also, here is the link to the RDE setup for reference. (https://github.com/SreejithNREL/2D_RDE/tree/withinletplenum)

-Sree

**Reply by RSuryaNarayan** [reply]
@SreejithNREL that is super helpful!!  thank you so much!

**Reply by marchdf** [reply]
@SreejithNREL can you push that to main pele?

**Reply by RSuryaNarayan** [reply]
@SreejithNREL I get a 404 error when I click the link to the 2D_RDE case. Could you please reshare that one?

**Reply by SreejithNREL** [reply]
Sorry, had kept that as private. I have added you as a collaborator. You should be able to access it now.

**Reply by RSuryaNarayan** [reply]
Thank you so much @SreejithNREL I can access it now.

**Reply by Pandakii** [reply]
Hi @SreejithNREL, could I refer to your RDE setup? I am very interested in some details of this issue, but I have no access to open your link https://github.com/SreejithNREL/2D_RDE/tree/withinletplenum.

**Reply by SreejithNREL** [reply]
Hi Pandakii, I have added you to the repo. Please let me know if you can access it. Thx

**Reply by Pandakii** [reply]
Thank you very much @SreejithNREL. I can  access it now.

**Reply by FernandoIGuerra** [reply]
Hi @SreejithNREL , Would you share your repo with me please? Also, When this mask is activated, the reaction at the remarked region is turned off and only in PeleC.cpp, peleC.h and React.cpp are made those changes? Thx.

**Reply by marchdf** [reply]
@SreejithNREL sounds like we are getting a lot of interest in this. Can you please make a PR so we can merge this?

**Reply by SreejithNREL** [reply]
The initial chemical mask implementation runs only on CPUs. I am currently evaluating it on GPUs and will make a PR once I complete some validations on GPUs. Meanwhile, I have added @FernandoIGuerra to my fork and he can start using/testing the feature.

@FernandoIGuerra : Please do the following:
Clone my PeleC fork. (https://github.com/SreejithNREL/PeleC.git)
Switch to “Reaction_mask_implementation_gpu” branch of PeleC.
cd to PelePhysics folder and checkout “Chem_Mask” branch of PelePhysics.

Please let me know (sreejith.nadakkalappukuttan@nrel.gov) if you run into issues.

**Reply by scramjetFoam** [reply]
@SreejithNREL I noticed you shared a tutorial on simulating 2D_RDE using PeleC, but now the repository seems to be private, is there any way to access it?

**Reply by SreejithNREL** [reply]
I have invited you to the repo. Please check and let me know if it does not work. Thx

**Reply by scramjetFoam** [reply]
> I have invited you to the repo. Please check and let me know if it does not work. Thx

It works. Thanks.

**Reply by Rachel008996** [reply]
@SreejithNREL  Could you please reshare the tutorial on simulating 2D_RDE using PeleC for me again? I would greatly appreciate it！ thank you very much！

**Reply by barkdullm99** [reply]
@SreejithNREL I am also currently struggling to get a stable RDE simulation. I have the chem_mask applied successfully, but suspect my initial conditions and boundary conditions are giving me errors. Would it be possible to share your 2D RDE repo with me as well?
